src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3902,6 +3902,14 @@ class WarehouseObjectPropertiesSegment(BaseSegment):
             Ref("EqualsSegment"),
             Ref("NakedIdentifierSegment"),
         ),
+        Sequence(
+            "ENABLE_QUERY_ACCELERATION",
+            Ref("EqualsSegment"),
+            Ref("BooleanLiteralGrammar"),
+            "QUERY_ACCELERATION_MAX_SCALE_FACTOR",
+            Ref("EqualsSegment"),
+            Ref("NumericLiteralSegment"),
+        ),
```

Comment:
Thanks for the PR! This is a good start.

I think these are distinct options. Their order shouldn't matter and `QUERY_ACCELERATION_MAX_SCALE_FACTOR` shouldn't require the `ENABLE_QUERY_ACCLERATION` option.
```suggestion
        Sequence(
            "ENABLE_QUERY_ACCELERATION",
            Ref("EqualsSegment"),
            Ref("BooleanLiteralGrammar"),
        ),
        Sequence(
            "QUERY_ACCELERATION_MAX_SCALE_FACTOR",
            Ref("EqualsSegment"),
            Ref("NumericLiteralSegment"),
        ),
```